nessie/src/main/java/org/apache/iceberg/nessie/NessieTableOperations.java

```diff
@@ -173,7 +207,7 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
       throw new CommitStateUnknownException(ex);
     } catch (NessieNotFoundException ex) {
       throw new RuntimeException(
-          String.format("Cannot commit: Reference '%s' no longer exists", client.getRef().getName()), ex);
+          String.format("Cannot commit: Reference '%s' no longer exists", updateableReference.getName()), ex);
```

Comment:
nit: I would probably revert all of those changes as I don't think they are necessary

Reply:
The result of `client.getRef()` _could_ change in the meantime.